### ISSUE TITLE
[14.0] QWeb templates not available in Frontend Portal pages for OWL components

### ISSUE THREAD

**Issue by PieterPaulussen** [odoo/odoo#73467]
Impacted versions:
14.0
Steps to reproduce:
1. Create a new OWL component with a static template reference.
2. Create a new QWeb template in static/src/xml/qweb_templates.xml and add it in the module manifest.
3. Build a new webpage with base template 'portal.portal_layout' and add a trigger to insert the OWL component.
4. Load the page with the widget, you get an error stating that the template could not be found.
Current behavior:
The qweb template it not available for the OWL component.
Expected behavior:
The qweb template is available for the OWL component
Video/Screenshot link (optional):

Support ticket number submitted via odoo.com/help (optional):

**Comment by pedrobaeza** [comment]
Not sure if this is an issue per se, or simply a limitation. @ged-odoo can you please answer?

**Comment by ged-odoo** [comment]
Yes, using owl in the frontend is not really easy for now (v14), because one needs to load templates manually. We should have some documentation/examples. We are currently working on updating the website to the new wowl framework and it will be much easier. A lot of work has been done on the asset management aspect of odoo, including the way we can now bundle templates (before, it was hardcoded in the `qweb` key, but we need different set of templates in different contexts, such as the web client, the point of sale or the website).

So, in short, I think that this is working as intended (if one needs to use owl in website, then it should load the templates manually (or inline it with the owl.tags.xml)), and starting in 15.0, the experience will be much better

**Comment by PieterPaulussen** [comment]
Thanks for the quick response! @ged-odoo and @pedrobaeza

**Comment by tarekmokhtarjr** [comment]
@PieterPaulussen 
Hello Pieter,

I had this issue too and i developed a small function to load qweb template (with trigger to insert the OWL component) from xml files (the same way it's done for the backend OWL based components).
Check this example:
https://gitlab.com/atarekmokhtar90/odoo_tips/-/tree/main/load_owl_template_from_xml_frontend

I tested it on Odoo v15

**Comment by flotho** [comment]
Hi @tarekmokhtarjr , thanks for your proposal.
Do you confirm that it's useless for  >=v16 version? Mounting a component with a target is really easy now.
